addons/point_of_sale/static/src/js/Screens/ProductScreen/ProductScreen.js

```diff
@@ -220,29 +225,25 @@ odoo.define('point_of_sale.ProductScreen', function(require) {
         async _showDecreaseQuantityPopup() {
             const { confirmed, payload: inputNumber } = await this.showPopup('NumberPopup', {
                 startingValue: 0,
-                title: this.env._t('Decrease the quantity by'),
+                title: this.env._t('Set the new quantity'),
             });
-            let decreaseQuantity = Math.abs(inputNumber);
-            if (confirmed && decreaseQuantity) {
+            let newQuantity = inputNumber !== ""? Math.abs(inputNumber): null;
+            if (confirmed && newQuantity !== null) {
                 let order = this.env.pos.get_order();
                 let selectedLine = this.env.pos.get_order().get_selected_orderline();
+                let currentQuantity = selectedLine.get_quantity()
 
-                let quantity = 0;
-                order.get_orderlines().forEach(orderLine => {
-                    if(orderLine.get_product().id === selectedLine.get_product().id && orderLine.get_discount() === selectedLine.get_discount())
-                        quantity += orderLine.get_quantity();
-                });
+                if(currentQuantity === 1 && newQuantity > 0)
```

Comment:
Is it ok for Lucie? shouldn't we also check that we are on the Last line? to avoid modification of previous lines?

Reply:
The check is done before, the function is only called if this is not the last line or if we want to reduce quantity.
And Lucie hasn't check this yet.

Reply:
ok cool